src/main/java/lotto/model/Reward.java

```diff
@@ -31,13 +28,13 @@ private int getMatchCount() {
         return this.matchCount;
     }
 
-    public static Reward getReward(WinningState winningState) {
-        if (winningState.isSpecificPrizeState(SECOND_PRIZE_COUNT_CONDITION, SECOND_PRIZE_BONUS_BALL_CONDITION)) {
+    public static Reward getReward(int matchCount, boolean bonusBall) {
+        if (matchCount == SECOND_PRIZE.matchCount && bonusBall == SECOND_PRIZE.bonusBall) {
             return SECOND_PRIZE;
         }
 
         return Arrays.stream(Reward.values())
-                .filter(reward -> reward.getMatchCount() == winningState.getMatchedCount())
+                .filter(reward -> reward.getMatchCount() == matchCount)
```

Comment:
```suggestion
                .filter(reward -> reward.matchCount == matchCount)
```
위와 같이 변경할 수 있을 것 같아요 😃